clean_build.sh

```diff
@@ -1,32 +1,74 @@
 #!/bin/sh
+
 cd "$(dirname "$0")" || exit 1
+
 # some environments have cmake v2 as 'cmake' and v3 as 'cmake3'
 # check for cmake3 first then fallback to just cmake
-B_CMAKE=`type cmake3 2>/dev/null`
-if [ $? -eq 0 ]; then
-    B_CMAKE=`echo "$B_CMAKE" | cut -d' ' -f3`
+
+B_CMAKE=$(command -v cmake3 2>/dev/null)
+if [ "$?" -eq 0 ]; then
+    # Continue, cmake3 exists in $PATH.
+    continue
 else
-    B_CMAKE=cmake
+    # OK, so cmake3 isn't in path, so let's test to see if `cmake` itself exists, before proceeding.
+    if command -v cmake 2>/dev/null; then
+        B_CMAKE=$(command -v cmake)
+        # We have a cmake executable available, now let's proceed!
+    else
+        # As self-explanatory, the cmake executable isn't available, so we should fail here.
+        echo "ERROR: CMake not in $PATH, cannot build! Please install CMake, or if this persists, file a bug report."
+        exit 1
+    fi
 fi
-# default build configuration
+
+# Set default build type, but in CI, this is set to Release.
+# For downstream distributions, our recommendation is to set this type in your build scripts to Release, and supply Debug build types in your debug packages.
+
 B_BUILD_TYPE=${B_BUILD_TYPE:-Debug}
+
 if [ "$(uname)" = "Darwin" ]; then
-    # OSX needs a lot of extra help, poor thing
-    # run the osx_environment.sh script to fix paths
-    . ./osx_environment.sh
-    B_CMAKE_FLAGS="-DCMAKE_OSX_SYSROOT=$(xcode-select --print-path)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 $B_CMAKE_FLAGS"
+    # macOS needs a little help, so we source this environment script to fix paths.
+    if [ ! -e "./macOS_environment.sh" ]; then
+        echo "macOS environment script not found, this isn't meant to happen!"
+        exit 1
+    else
+        . ./macOS_environment.sh
```

Comment:
Executing this will fail if there's no file, so there's no need for the check of existence. I doubt that in normal times we'll ever see the above error ever happen.

Reply:
Also fixed.